var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1242,6 +1267,35 @@ def setup_dependent_package(self, module, dependent_spec):
         module.python_platlib = join_path(dependent_spec.prefix, self.platlib)
         module.python_purelib = join_path(dependent_spec.prefix, self.purelib)
 
+    def add_files_to_view(self, view, merge_map, skip_if_exists=True):
+        """Make the view a virtual environment if it isn't one already.
+
+        If `python-venv` is linked into the view, it will already be a virtual
+        environment. If not, then this is an older python that doesn't use the
+        python-venv support, or we may be using python packages that
+        use ``depends_on("python")`` but not ``extends("python")``.
+
+        We used to copy the python interpreter in, but we can get the same effect in a
+        simpler way by adding a ``pyvenv.cfg`` to the environment.
+
+        """
+        super().add_files_to_view(view, merge_map, skip_if_exists=skip_if_exists)
+
+        # location of python inside the view, where we will put the venv config
+        projection = view.get_projection_for_spec(self.spec)
+        pyvenv_cfg = os.path.join(projection, "pyvenv.cfg")
+        if os.path.lexists(pyvenv_cfg):
+            return
+
+        # if this is a copy view, the realpath of python will be in the view
+        python_path = find_python_in_prefix(self.spec, projection).path
+        copy_view = not os.path.commonpath((os.path.realpath(python_path), projection))
```

Comment:
If you use realpath you should use it on both lhs an rhs, otherwise it breaks when the view itself has a symlinked parent dir.

Also `not os.path.commonpath(...)` is always false since it is at least `/`.

Maybe expose the view type from the view object instead of inferring it?

Reply:
Sigh. I wanted to but tried to avoid it b/c it is a bigger change. I think it's the best thing to do though.

Reply:
ok this is done.